lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -778,6 +935,20 @@ createAsyncOps(scf::ForOp &forOp,
   if (phase) {
     phase = newForOp.getBody()->getArgument(newOperandIndex + 2);
   }
+  auto tmaCounters = ArrayRef<BlockArgument>(newForOp.getBody()->getArguments())
+                         .slice(newOperandIndex + (phase ? 3 : 2));
+
+  // Update yield op with temporary yield values
+  auto forYield = cast<scf::YieldOp>(newForOp.getBody()->getTerminator());
+  for (unsigned i = 0; i < newOperands.size(); ++i) {
+    forYield.getResultsMutable().append(newOperands[i]);
+  }
+
+  if (failed(rewriteTMABufferUpdates(newForOp, tmaBufferMapping, tmaCounters,
+                                     numBuffersVal, one, zero))) {
```

Comment:
We have discussed this with Peter elsewhere - to be safe we should be even more conservative regarding number of buffers for descriptors - we should be probably using `numStages` instead of `numBuffers` here.